plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/UserMethodSignatureEmbedding.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.embeddings
+
+import org.jetbrains.kotlin.formver.conversion.ReturnVariableName
+import org.jetbrains.kotlin.formver.viper.ast.*
+
+interface UserMethodSignatureEmbedding : MethodSignatureEmbedding {
+    val receiver: VariableEmbedding?
+    val params: List<VariableEmbedding>
+
+    val returnVar
+        get() = VariableEmbedding(ReturnVariableName, returnType)
+    val formalArgs: List<VariableEmbedding>
+        get() = listOfNotNull(receiver) + params
+
+    override val receiverType: TypeEmbedding?
+        get() = receiver?.type
+    override val paramTypes: List<TypeEmbedding>
+        get() = params.map { it.type }
+}
+
+fun UserMethodSignatureEmbedding.toMethodCall(
```

Comment:
This method can be part of `MethodSignatureEmbedding`.
Nit: I would also move it into the interface rather than define it as extension.

Reply:
Oh, I missed that it doesn't need the parameter names, good point.

My logic for making it an extension was that it does not make any sense to override it; I feel like putting it in the interface would suggest there may be a use for that.  (Perhaps the same could be said of `formalArgs` and `formalArgTypes`, not sure.)